File: GGAndroid_2.0/GGOpen/open-wechat/src/main/java/io/ganguo/wechat/WeChatActivity.java
package io.ganguo.wechat;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;

import com.tencent.mm.opensdk.constants.ConstantsAPI;
import com.tencent.mm.opensdk.modelbase.BaseReq;
import com.tencent.mm.opensdk.modelbase.BaseResp;
import com.tencent.mm.opensdk.modelmsg.SendAuth;
import com.tencent.mm.opensdk.openapi.IWXAPIEventHandler;

import io.ganguo.utils.common.ResHelper;
import io.ganguo.wechat.callback.WeChatCallBack;

/**
 * 微信分享、认证回调Activity
 * Created by zoyen on 2018/7/10.
 */
public class WeChatActivity extends Activity implements IWXAPIEventHandler {
    public static String code;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        WeChatManager.iwxapi.handleIntent(getIntent(), this);
    }

    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);
        setIntent(intent);
        WeChatManager.iwxapi.handleIntent(intent, this);
    }

    @Override
    public void onReq(BaseReq baseReq) {

    }

    @Override
    public void onResp(BaseResp baseResp) {
        WeChatCallBack callback = WeChatManager.getCallback();
        if (callback != null) {
            onFilterResult(baseResp, callback);
        }
        WeChatManager.setCallback(null);
        finish();
    }

    /**
     * function: 对返回结果做过滤处理
     *
     * @param baseResp
     */
    protected void onFilterResult(BaseResp baseResp, WeChatCallBack callBack) {
        switch (baseResp.errCode) {
            case BaseResp.ErrCode.ERR_OK:
                onResultSuccess(baseResp, callBack);
                break;
            case BaseResp.ErrCode.ERR_USER_CANCEL:
                callBack.onCancel();
                break;
            default:
                onResultFailed(baseResp, callBack);
                break;
        }
    }

    /**
     * function: 分享/认证成功
     *
     * @param baseResp
     * @param callBack
     */
    protected void onResultSuccess(BaseResp baseResp, WeChatCallBack callBack) {
        if (baseResp.getType() == ConstantsAPI.COMMAND_SENDAUTH) {
            callBack.onSuccess(((SendAuth.Resp) baseResp).code);
        } else {
            callBack.onSuccess(ResHelper.getString(R.string.str_wechat_share_success));
        }
    }


    /**
     * function: 分享/认证成功
     *
     * @param baseResp
     * @param callBack
     */
    protected void onResultFailed(BaseResp baseResp, WeChatCallBack callBack) {
        if (baseResp.getType() == ConstantsAPI.COMMAND_SENDAUTH) {
            callBack.onFailed(new RuntimeException(ResHelper.getString(R.string.str_wechat_auth_failed) + BaseResp.ErrCode.ERR_AUTH_DENIED));
        } else {
            callBack.onFailed(new RuntimeException(ResHelper.getString(R.string.str_wechat_share_failed) + BaseResp.ErrCode.ERR_AUTH_DENIED));
        }
    }

}

File: GGAndroid_4.0/core/service/service-open/open-wechat/src/main/kotlin/io/ganguo/wechat/WeChatActivity.kt
package io.ganguo.wechat

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.util.Log
import com.tencent.mm.opensdk.modelbase.BaseReq
import com.tencent.mm.opensdk.modelbase.BaseResp
import com.tencent.mm.opensdk.openapi.IWXAPI
import com.tencent.mm.opensdk.openapi.IWXAPIEventHandler
import com.tencent.mm.opensdk.openapi.WXAPIFactory
import io.ganguo.wechat.helper.WXOpenHelper


/**
 * <pre>
 * author : zoyen
 * time   : 2018/7/10
 * desc   : 微信分享、认证回调Activity
 */
open class WeChatActivity : Activity(), IWXAPIEventHandler {
    private var iwxapi: IWXAPI? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        iwxapi = WXAPIFactory.createWXAPI(this, WXConstants.WX_APP_ID)
        val handled = iwxapi!!.handleIntent(intent, this)
        if (!handled) {
            finish()
        }
    }

    override fun onNewIntent(intent: Intent) {
        super.onNewIntent(intent)
        setIntent(intent)
        if (iwxapi != null) {
            iwxapi!!.handleIntent(intent, this)
        }
    }

    override fun onReq(baseReq: BaseReq) {

    }

    /**
     * 微信sdk结果回调函数
     */
    override fun onResp(resp: BaseResp) {
        Log.d("onResp", "WeChat Open Sdk onResp errCode==${resp.errCode},type==${resp.type}")
        WXOpenHelper.onResult(resp)
        finish()
    }

    override fun onDestroy() {
        iwxapi = null
        super.onDestroy()
    }


}
